sycl/include/CL/sycl/ONEAPI/sub_group.hpp

```diff
@@ -224,6 +229,50 @@ struct sub_group {
 
   /* --- sub_group load/stores --- */
   /* these can map to SIMD or block read/write hardware where available */
+#ifdef __SYCL_DEVICE_ONLY__
+  // Method for decorated pointer
+  template <typename T>
+  detail::enable_if_t<
+      !std::is_same<typename detail::remove_AS<T>::type, T>::value, T>
+  load(T *src) const {
+    return load(sycl::multi_ptr<typename detail::remove_AS<T>::type,
+                                sycl::detail::deduce_AS<T>::value>(
+        (typename detail::remove_AS<T>::type *)src));
+  }
+
+#ifndef SYCL_USE_DECORATED_REF
+  // Method for raw pointer
+  template <typename T>
+  detail::enable_if_t<
+      std::is_same<typename detail::remove_AS<T>::type, T>::value, T>
+  load(T *src) const {
+
+    auto p = __spirv_GenericCastToPtrExplicit_ToPrivate<T>(
+        src, __spv::StorageClass::Function);
+    if (p)
+      return load(p);
```

Comment:
Given my comment above, we should remove this.

I don't know if it should be an assert or if we should just let it go through to the fallback (as undefined behavior).  An assert would be nice, but only if we can prove that it doesn't prevent the compiler from optimizing away the other runtime checks in cases where the address space is known.

Reply:
done
added assert if cast to private succeeded